server/app/services/applicant/ReadOnlyApplicantProgramServiceImpl.java

```diff
@@ -84,6 +85,17 @@ public ImmutableList<String> getStoredFileKeys() {
         .collect(ImmutableList.toImmutableList());
   }
 
+  @Override
+  public boolean isBlockEligible(String blockId) {
+    Block block = getBlock(blockId).get();
+    Optional<PredicateDefinition> predicate =
+        block.getEligibilityDefinition().map(EligibilityDefinition::predicate);
+    if (predicate.isEmpty()) {
+      return true;
+    }
+    return evaluateEligibility(block, predicate.get());
```

Comment:
suggestion (non-blocking): will java let you do this?
```suggestion
    return  predicate.isEmpty() || evaluateEligibility(block, predicate.get());
```

Reply:
I believe so, java does lazy evaluation.  I feel like this hurts readability though as the two evaluations are not evaluating the same things which can then be error prone.